### PR TITLE
Cache kubeconfig lookups

### PR DESCRIPTION
**Pull request by tobiscr**

### CODE REVIEW

File: pkg/cluster/inventory.go [hunk 1]
```diff
@@ -57,6 +65,19 @@ func NewInventory(conn db.Connection, debug bool, collector metricsCollector) (I
 		return nil, err
 	}
 
+	//ensure the cache get's initialized just once
+	mutex.Lock()
+	if kcCache == nil {
+		kcCache = ttlcache.New[string, string](
+			ttlcache.WithTTL[string, string](maxTTL),
```
Comment (jeremyharisch) [inline pkg/cluster/inventory.go:72]: You need to disable the touch, otherwise TTL will always be renewed and never triggered

Comment (tobiscr) [reply pkg/cluster/inventory.go:72]: aaah, nice catch man! :) 

https://github.com/jellydator/ttlcache/blob/v3/options.go#L73

**Comment by tobiscr** [comment]
Closing are replaced by #1433

**Pull request closed without merging**